src/renderers/shared/fiber/ReactFiberCompleteWork.js

```diff
@@ -132,6 +130,17 @@ module.exports = function<T, P, I, C>(config : HostConfig<T, P, I, C>) {
         return null;
       case ClassComponent:
         transferOutput(workInProgress.child, workInProgress);
+        // Don't use the state queue to compute the memoized state. We already
+        // merged it and assigned it to the instance. Transfer it from there.
+        const state = workInProgress.stateNode.state;
+        workInProgress.memoizedState = state;
+        // Transfer update queue to callbackList field so callbacks can be
+        // called during commit phase.
+        workInProgress.callbackList = workInProgress.updateQueue;
+        if (current) {
+          current.callbackList = workInProgress.callbackList;
```

Comment:
Ah alright. I'll need to rethink what to do here then: https://github.com/acdlite/react/blob/04c9a399e871d5ffe7323f11132b51185bc32e6e/src/renderers/shared/fiber/ReactFiberBeginWork.js#L158-L174

Does the overall strategy of callback list seem sound or should I go back to the drawing board?